os/android/platformdefines.h

```diff
@@ -54,44 +55,24 @@ struct gralloc_handle {
 };
 
 typedef struct gralloc_handle* HWCNativeHandle;
-typedef struct cros_gralloc_handle HWCNativeBuffer;
-#define GETNATIVEBUFFER(handle) (*(cros_gralloc_handle*)(handle->handle_))
-
-
-struct BufferHash {
-  size_t operator()(HWCNativeBuffer const& buffer) const {
-    const native_handle_t & p = buffer.base;
-    std::size_t seed = 0;
-    for (int i = 0; i < p.numFds + p.numInts; i++) {
-      hash_combine_hwc(seed, (const size_t)p.data[i]);
-    }
-    return seed;
-  }
-};
-
-struct BufferEqual {
-  bool operator()(const HWCNativeBuffer& buffer1, const HWCNativeBuffer& buffer2) const {
-    const native_handle_t &p1 = buffer1.base;
-    const native_handle_t &p2 = buffer2.base;
-    bool equal = (p1.numFds == p2.numFds) && (p1.numInts ==  p2.numInts);
-    if (equal) {
-      for (int i = 0; i < p1.numFds + p1.numInts; i++) {
-        equal = equal && (p1.data[i] == p2.data[i]);
-        if (!equal)
-          break;
-      }
-    }
-    return equal;
-  }
-};
-
 
 #define VTRACE(fmt, ...) ALOGV("%s: " fmt, __func__, ##__VA_ARGS__)
 #define DTRACE(fmt, ...) ALOGD("%s: " fmt, __func__, ##__VA_ARGS__)
 #define ITRACE(fmt, ...) ALOGI(fmt, ##__VA_ARGS__)
 #define WTRACE(fmt, ...) ALOGW("%s: " fmt, __func__, ##__VA_ARGS__)
 #define ETRACE(fmt, ...) ALOGE("%s: " fmt, __func__, ##__VA_ARGS__)
 #define STRACE() ATRACE_CALL()
+
+inline uint32_t GetNativeBuffer(uint32_t gpu_fd, HWCNativeHandle handle)
+{
+  uint32_t id = 0;
+  uint32_t prime_fd = handle->handle_->data[0];  
+  if (drmPrimeFDToHandle(gpu_fd, prime_fd, &id)) {
```

Comment:
hmm will this not change for every call ? do we hit cache ever ? also, don't we need to close the fds.

Reply:
hmm overall seems to work fine. This is my only concern, dont we need to close this id here.

Reply:
id is the same with imported handle. id is the only reliable identification for buffer. When you release graphics buffer, already told kernel the id is not used anymore

Reply:
@kalyankondapally  I removed the video handle=0 because it was a WA before. I removed those 2 lines and I don't see any warnings and errors. In theory what I am doing here is right